typescript/tsconfig.base.json

```diff
@@ -0,0 +1,8 @@
+{
+  "$schema": "https://json.schemastore.org/tsconfig",
+  "extends": "@tsconfig/recommended",
```

Comment:
Note this dependency is not "official". Its maintained by Orta who no longer works on TS.

I actually liked keeping it explicit since our `@vercel/style-guide` is already a recommended base.

Reply:
Oh true! Okay I can agree with that. We should document our rule choices then so that we don't lose context for _why_ we made these decisions.